nixos/modules/services/networking/routinator.nix

```
@@ -0,0 +1,188 @@
+            '';
+            default = null;
+          };
+          refresh = mkOption {
+            type = types.nullOr types.int;
+            description = ''
+              An integer value specifying the number of seconds Routinator should wait between consecutive validation runs in server mode. The next validation run will happen earlier, if objects expire earlier.
+            '';
+            default = 600;
+          };
+          retry = mkOption {
+            type = types.nullOr types.int;
+            description = ''
+              An integer value specifying the number of seconds an RTR client is requested to wait after it failed to receive a data set.
+            '';
+            default = 600;
+          };
+          expire = mkOption {
+            type = types.nullOr types.int;
+            description = ''
+              An integer value specifying the number of seconds an RTR client is requested to use a data set if it cannot get an update before throwing it away and continuing with no data at all.
+            '';
+            default = 7200;
+          };
+        };
+      };
+      description = ''
+        Configuration for Routinator 3000, see <https://routinator.docs.nlnetlabs.nl/en/stable/manual-page.html#configuration-file> for options.
+      '';
+      default = { };
+    };
+  };
+
+  config = {
+    environment.etc."routinator.conf".source = settingsFormat.generate "routinator.conf" (
+      filterAttrsRecursive (n: v: v != null) cfg.settings
+    );
+
+    systemd.services.routinator = {
+      description = "Routinator 3000 is free, open-source RPKI Relying Party software made by NLnet Labs.";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+      restartTriggers = [ config.environment.etc."routinator.conf".source ];
+      path = with pkgs; [ rsync ];
+      serviceConfig = {
+        Type = "exec";
+        ExecStart = escapeSystemdExecArgs (
+          [
+            (getExe cfg.package)
+            "--config=/etc/routinator.conf"
+          ]
+          ++ cfg.extraArgs
+          ++ [
+            "server"
+          ]
+          ++ cfg.extraServerArgs
+        );
+        Restart = "on-failure";
+        DynamicUser = true;
+        LockPersonality = true;
+        MemoryDenyWriteExecute = true;
+        NoNewPrivileges = true;
+        PrivateDevices = true;
+        PrivateTmp = true;
+        ProtectClock = true;
+        ProtectControlGroups = true;
+        ProtectHome = true;
+        ProtectHostname = true;
+        ProtectKernelModules = true;
+        ProtectKernelTunables = true;
+        ProtectSystem = "strict";
+        RestrictAddressFamilies = "AF_UNIX AF_INET AF_INET6";
+        RestrictNamespaces = true;
+        RestrictRealtime = true;
+        StateDirectory = "routinator";
+        SystemCallArchitectures = "native";
+        SystemCallErrorNumber = "EPERM";
+        SystemCallFilter = "@system-service";
```

```suggestion
        SystemCallFilter = "@system-service";
        UMask = "0027";
```

which maps to 0750/0640 in practice